### PR TITLE
File open feature

### PR DESCRIPTION
**Pull request by ScottDemarest**
This is a redo of the previous pull request. Used flake8, should pass tests this time.

### CODE REVIEW

**Comment by ScottDemarest** [comment]
I fixed most of the typing issues, but there are some I do not know how to resolve. We'll talk about this later

**Comment by ScottDemarest** [comment]
yay

**Comment by nsmith-** [comment]
Ok the remaining CI failure seems to be a real issue

**Review by nsmith- — CHANGES_REQUESTED** [review]

File: src/fsspec_xrootd/xrootd.py [hunk 1]
```diff
@@ -1 +1,268 @@
...
+        if mode == "rb":
+            if size is not None:
+                self.size = size
+            else:
+                self.size = self.details["size"]
+            self.cache = caches[cache_type](
+                self.blocksize, self._fetch_range, self.size, **cache_options
+            )
+        else:
+            self.buffer = io.BytesIO()
```
[the remark this answers is not in the document]
Comment (ScottDemarest) [reply src/fsspec_xrootd/xrootd.py:240]: Its used in the constructor by default. We dont need it for this branch, but will for the write operations

File: tests/File_Read_Feature_Tests.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (ScottDemarest) [reply tests/File_Read_Feature_Tests.py:1]: my local precommit put that line there. I will remove it for now, but precommit will put it back each time I run it

[the remark this answers is not in the document]
Comment (nsmith-) [reply tests/File_Read_Feature_Tests.py:1]: I mean `git rm --cached` the whole file

[the remark this answers is not in the document]
Comment (ScottDemarest) [reply tests/File_Read_Feature_Tests.py:1]: Oh, wait, you're right. I'll delete it. Also, do we even need this file? Your test file does everything this one does

[the remark this answers is not in the document]
Comment (ScottDemarest) [reply tests/File_Read_Feature_Tests.py:1]: I removed the file

**Pull request merged**